consensus/istanbul/backend/engine.go

```diff
@@ -443,95 +389,166 @@ func (sb *Backend) UpdateValSetDiff(chain consensus.ChainReader, header *types.H
 	return nil
 }
 
-// TODO(brice): This needs a comment.
+// Returns whether or not a particular header represents the last block in the epoch.
 func (sb *Backend) IsLastBlockOfEpoch(header *types.Header) bool {
 	return istanbul.IsLastBlockOfEpoch(header.Number.Uint64(), sb.config.Epoch)
 }
 
+// Returns the size of epochs in blocks.
+func (sb *Backend) EpochSize() uint64 {
+	return sb.config.Epoch
+}
+
 // Finalize runs any post-transaction state modifications (e.g. block rewards)
 // and assembles the final block.
 //
 // Note, the block header and state database might be updated to reflect any
 // consensus rules that happen at finalization (e.g. block rewards).
 func (sb *Backend) Finalize(chain consensus.ChainReader, header *types.Header, state *state.StateDB, txs []*types.Transaction, uncles []*types.Header, receipts []*types.Receipt, randomness *types.Randomness) (*types.Block, error) {
+	header.Root = state.IntermediateRoot(chain.Config().IsEIP158(header.Number))
```

Comment:
Looks like all of the errors will be ignored in this method.  Is that intended?

Reply:
As we discussed, it does appear that the contract_call library functions will output an evm error if it's encountered.